### DISCUSSION TITLE
QR Code Printing Improvements

### DISCUSSION THREAD

**Discussion by danricho** [danricho/where_the#10] [Ideas]
- Add batch printing of QR Codes
- Add templating for printing onto pre-cut adhesive labels (eg. Avery Template)

**Comment by poblabs** [comment]
I would also like to see an option to remove logo from center. as well as ability to add text to the code

**Reply by danricho** [reply]
I could do logo configuration quite easily (via the config.yml). Regarding the text, where would you input that text? 

Would it be stored against the item or just added to the printed QR code?

The biggest issue with the text for me would be handling too much text...

**Reply by danricho** [reply]
I have added a configuration option for disabling the logo in the QR code (a2b3284).

**Reply by poblabs** [reply]
Not sure about the text positioning. I'd have to experiment, but I was thinking that the location ID doesn't tell me much on what the QR code is for. I mean - I know what it is now... but in 6 months 🤔 😄

**Reply by danricho** [reply]
I guess that is the main reason you would use the QR code... it links directly to the view page for that box/shelf/location.

**Reply by poblabs** [reply]
True, but perhaps instead of "MK120", it could be the location title, like "Basement Box" or whatever?

**Reply by danricho** [reply]
Perhaps an option to create a custom ID would suffice, rather than it generating random IDs?
The main reason it is random is so that there are a very large number of possibilities and the function that makes them also checks for conflicts. Below is where the ID is created.

https://github.com/danricho/where_the/blob/aeb2b9967f531591b5ab6726041a4d1cc5d546d7/main.py#L93 

I'd welcome some more of your thoughts on this, but consideration on how to make the QR code format work for any new solution as well as uniqueness.

**Reply by poblabs** [reply]
I'm fine with the random ID for the location itself, but on the QR code I think it can be forgotten in future (agreed, you could scan it to remind yourself). My thought was maybe just an option on the QR code to change that text with something else. If it's not changed then use the location ID as is

**Reply by danricho** [reply]
In that case we are only talking about the print jinja template…. Maybe some small text under the QR code that is the first x characters of the description? Or just use those characters in lieu of the ID as you say… would 6 be enough though? It becomes a question of how big the label should be.

I’ll give a few things a try when I’m next working on it. Thanks for the feedback!

**Reply by danricho** [reply]
@poblabs I have added a configuration option for the description under the QR code (8f4d74c)

Note that a really long description may end up looking pretty bad. 

Hope that helps.

**Comment by danricho** [comment]
Added batch printing of QR Codes and templating for printing onto pre-cut adhesive labels (eg. Avery Template) (bde1f2a)
